### PR TITLE
MDS-3330/MDS-3451/MDS-3476/MDS-3472: Various Admin-Side Tweaks

### PR DESCRIPTION
**Pull request by exzizt**
# Main
## MDS-3330
* Adds the well authorization numbers beside each payment detail and as a memo in generated PRFs
image
## MDS-3451
* Updates the qualified receiver name list displayed on PRFs
image
## MDS-3476
* In the modal to update statuses on contracted work payments, shows a warning message if the interim EOC total and final EOC total are the same or if the final EOC total is greater than the interim EOC total
image
image
## MDS-3472
* The selected contracted work payment items are now deselected when a PRF is generated

# Other
* Changes the filtering behaviour for interim and final statuses on contracted work payments to be "and" instead of "or"
* Adds work_id to the work_info_view

### CODE REVIEW

File: services/dsrp-api/app/api/application/models/payment_document.py [hunk 1]
```diff
@@ -48,11 +48,19 @@ def create_invoice_number(application):
             invoice_number = f'{agreement_number}-{payment_phase}-{amount_generated + 1}'
             return invoice_number
 
+        def get_memo():
+            well_authorization_numbers = []
+            for cwp in self.contracted_work_payments:
+                cw = self.application.find_contracted_work_by_id(cwp.work_id)
+                well_authorization_numbers.append(cw['well_authorization_number'])
```
Comment (mhuseinov) [inline services/dsrp-api/app/api/application/models/payment_document.py:57]: Do we need any 'None' check at this point?

Comment (exzizt) [reply services/dsrp-api/app/api/application/models/payment_document.py:57]: It should 100% exist but might not be bad to just check anyway and throw an exception ;)

**Review by mhuseinov — APPROVED** [review]
🔥

**Review by mhuseinov — APPROVED** [review]
💯

**Pull request merged**